src/nuggetizer/core/llm.py

```diff
@@ -133,49 +141,104 @@ def run(
             try:
                 completion = None
                 # Use different parameters for vLLM vs other APIs
+                completion_params: Dict[str, Any]
                 if hasattr(self.client, "base_url") and "localhost" in str(
                     self.client.base_url
                 ):
                     # vLLM specific parameters
-                    completion = self.client.chat.completions.create(
-                        model=self.model,
-                        messages=messages,  # type: ignore[arg-type]
-                        temperature=temperature,
-                        max_tokens=4096,
-                        timeout=60,
-                    )
+                    completion_params = {
+                        "model": self.model,
+                        "messages": messages,
+                        "temperature": temperature,
+                        "max_tokens": 4096,
+                        "timeout": 60,
+                    }
                 else:
                     # Standard OpenAI/other APIs
-                    completion = self.client.chat.completions.create(
-                        model=self.model,
-                        messages=messages,  # type: ignore[arg-type]
-                        temperature=temperature,
-                        max_completion_tokens=4096,
-                        timeout=60,
-                    )
+                    completion_params = {
+                        "model": self.model,
+                        "messages": messages,
+                        "temperature": temperature,
+                        "max_completion_tokens": 4096,
+                        "timeout": 60,
+                    }
+                completion = self.client.chat.completions.create(**completion_params)
+                # print(f"🔍 DEBUG LLM: API call completed successfully") # not
+                # removed because it's very helpful for debugging
```

Comment:
1) I think we should remove these even if it is helpful for debugging, coding practice wise it is for the developer to debug.
2) We could have a --debug mode where we expose such things, but that is a separate PR. And we shouldn't be doing print within function methods, logging is a better alternative.